File: SQLiteRepository/SQLiteQuery/InsertQuery.cs
using System.Collections.Generic;
using System.Data.SQLite;

namespace SQLiteRepository.SQLiteQuery
{
    public class InsertQuery : ISqliteQuery 
    {
        public string TableName { get; set; }
        public List<List<string>> Values { get; set; }
        public string Path { get; set; }
        public List<string> Columns { get; set; }
        public IQueryResult Execute()
        {
            return InsertSqlite.Insert(this);
        }
        public IQueryResult Execute(SQLiteConnection conn, SQLiteTransaction tr)
        {
            return InsertSqlite.Insert(conn,tr, this);
        }
    }
}

File: SQLiteAdapters/InsertQuery.vb
Imports System.Data.SQLite

Public Class InsertQuery
    Inherits BaseQuery

    Public Property Values As List(Of List(Of String))

    Public Overrides Function Execute() As IQueryResult
        Return InsertSqlite.Insert(Me)
    End Function
    Public Overrides Function Execute(conn As SQLiteConnection, tr As SQLiteTransaction) As IQueryResult
        Return InsertSqlite.Insert(conn, tr, Me)
    End Function
End Class